datumaro/components/format_detection.py

```diff
@@ -180,6 +200,80 @@ def probe_text_file(
         except Exception:
             self.fail(requirement_desc_full)
 
+    @contextlib.contextmanager
+    def alternatives(self) -> Iterator[None]:
+        """
+        Returns a context manager that can be used to place a requirement that
+        is considered met if at least one of several alternative sets of
+        requirements is met.
+        To do so, use a `with` statement, with the alternative sets of
+        requirements represented as nested `with` statements using the context
+        manager returned by `alternative`:
+
+            with context.alternatives():
+                with context.alternative():
+                    # place requirements from alternative set 1 here
+                with context.alternative():
+                    # place requirements from alternative set 2 here
+                ...
+
+        The contents of all `with context.alternative()` blocks will be
+        executed, even if an alternative that is met is found early.
+
+        Requirements must not be placed directly within a
+        `with context.alternatives()` block.
+        """
+
+        self._start_requirement()
```

Comment:
Probably, this will produce a confusing error, if alternatives are used twice in a row.

Reply:
What do you mean? This?

```python
with context.alternatives():
   ...

with context.alternatives():
   ...
```

That shouldn't result in any error. (And it checks that at least one alternative from the first block succeeds _and_ at least one alternative from the second block succeeds.)

Reply:
No, I meant this:
```python
with context.alternatives():
  with context.alternatives():
    ...
```
This is going to produce `"checks can't be done directly within `alternatives` blocks"`, which is pretty misleading.

Reply:
Well... `alternatives` _is_ a check, but I see what you mean. I'll think about how I can improve this.

Reply:
Oh, I just realized that I've mixed up my terminology in these messages (I changed "checks" to "requirements" when I went from prototype to real implementation). I'll definitely have to fix that.

Reply:
Okay, I tuned the message and added the type of the requirement that was at fault. That should be clear enough.